pkg/pillar/cmd/zedrouter/cni.go

```diff
@@ -326,7 +327,7 @@ func (z *zedrouter) getAppByPodName(
 	}
 	for _, item := range z.pubAppNetworkStatus.GetAll() {
 		appStatus := item.(types.AppNetworkStatus)
-		if strings.ToLower(appStatus.DisplayName) == appName &&
+		if strings.ToLower(appStatus.DisplayName[:base.EveKubeAppMaxNameLen]) == appName &&
```

Comment:
This still would not match if `DisplayName` had e.g. underscores that we have replaced.

Reply:
Got it. updated and include checking general case.